File: problems/415/solution.go
package main
import "strconv"

func addStrings(num1 string, num2 string) string {
    n1l, n2l, ans, cur := len(num1) - 1, len(num2) - 1, "", false
    
    for n1l > -1 || n2l > -1 {
        ns := 0
        
        if n1l > -1 {
            ns += int(num1[n1l] - '0')
            n1l--
        }
        
        if n2l > -1 {
            ns += int(num2[n2l] - '0')
            n2l--
        }
        
        if cur {
            ns++
            cur = false
        }
        
        if ns > 9 {
            cur = true
            ns -= 10
        }
        
        ans = strconv.Itoa(ns) + ans
    }
    
    if cur {
        ans = "1" + ans
    }
    
    return ans
}

File: problems/415/solution.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        n1, n2, s, cur = len(num1) - 1, len(num2) - 1, "", False
        
        while n1 >= 0 or n2 >= 0:
            ns = 0

            if n1 >= 0:
                ns += (ord(num1[n1]) - ord('0'))
                n1 -= 1
            
            if n2 >= 0:
                ns += (ord(num2[n2]) - ord('0'))
                n2 -= 1
            
            if cur:
                ns += 1
                cur = False
            
            if ns > 9:
                cur = True
                ns -= 10
            
            s = str(ns) + s
        
        if cur:
            s = "1" + s
        
        return s